src/ui/wxWidgets/TreeCtrl.cpp

```diff
@@ -785,6 +819,46 @@ void TreeCtrl::OnKeyDown(wxTreeEvent& evt)
   evt.Skip();
 }
 
+void TreeCtrl::OnMouseRightClick(wxMouseEvent& event)
+{
+  wxPoint mouseClickPosition = event.GetPosition();
+  int positionInfo;
+
+  HitTest(mouseClickPosition, positionInfo);
+
+  if ((positionInfo | wxTREE_HITTEST_NOWHERE) == wxTREE_HITTEST_NOWHERE) {
+    auto *parentWindow = dynamic_cast<PasswordSafeFrame*>(GetParent());
+    wxASSERT(parentWindow != nullptr);
+    /*
+      Triggers an event that interferes with the popup menu in that sense that 
+      an invalid wxTreeItemId is provided at the moment when the popup menu shows.
+      This causes the menu items to get disabled at PasswordFrame::OnUpdateUI.
+      'Yield' solves this problem so that events are handled in the right order.
+    */
+    Unselect();
+    wxGetApp().SafeYieldFor(parentWindow, wxEVT_CATEGORY_USER_INPUT);
+    parentWindow->OnContextMenu(nullptr);
+  }
+  else {
+    event.Skip();
+  }
+}
+
+void TreeCtrl::OnMouseLeftClick(wxMouseEvent& event)
+{
+  wxPoint mouseClickPosition = event.GetPosition();
+  int positionInfo;
+
+  HitTest(mouseClickPosition, positionInfo);
+
+  if ((positionInfo | wxTREE_HITTEST_NOWHERE) == wxTREE_HITTEST_NOWHERE) {
```

Comment:
I don't know much about the wxWidgets API, but if `wxTREE_HITTEST_NOWHERE` is a mask with a single bit set to 1, shouldn't the operator be `&`?

Reply:
Agree - certainly looks like this should be "&", not "|"

Reply:
Ow, good catch. - Fixed.

Reply:
Fixed in OnMouseRightClick but not in OnMouseLeftClick.

Reply:
Thanks for pointing this out. Fixed in #609.